swarm/runtime/evolution.py

```
@@ -965,12 +965,15 @@ def list_pending_patches(
     if not runs_root.exists():
         return results
 
-    run_dirs = sorted(runs_root.iterdir(), reverse=True)[:limit]
+    import os
+    run_dirs = []
+    with os.scandir(runs_root) as entries:
+        sorted_entries = sorted(entries, key=lambda e: e.name, reverse=True)[:limit]
+        for entry in sorted_entries:
+            if entry.is_dir():
+                run_dirs.append(Path(entry.path))
```

Ruff (0.15.17)

[warning] Use `list.extend` to create a transformed list

(PERF401)